src/Mvc/Mvc.Formatters.Xml/src/XmlDataContractSerializerInputFormatter.cs

```diff
@@ -118,43 +117,55 @@ public override async Task<InputFormatterResult> ReadRequestBodyAsync(InputForma
             }
 
             var request = context.HttpContext.Request;
+            Stream readStream = new NonDisposableStream(request.Body);
 
```

Comment:
Why is this needed?

Reply:
And, why not `var`?

Reply:
This is existing code. Users may opt out of buffering, we still need to read it in that case. However we do not want to close the stream once the reader is done.

Reply:
We assign this to a `FileBufferingReadStream` in the `if`, so cannot be an instance of `NonDisposableStream`